modules/osdk-run-proxy.adoc

```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/golang/osdk-golang-tutorial.adoc
+// * operators/operator_sdk/ansible/osdk-ansible-tutorial.adoc
+// * operators/operator_sdk/helm/osdk-helm-tutorial.adoc
+
+ifeval::["{context}" == "osdk-golang-tutorial"]
+:golang:
+:app-proper: Memcached
+:app: memcached
+:group: cache
+endif::[]
+ifeval::["{context}" == "osdk-ansible-tutorial"]
+:ansible:
+:app-proper: Memcached
+:app: memcached
+:group: cache
+endif::[]
+ifeval::["{context}" == "osdk-helm-tutorial"]
+:helm:
+:app-proper: Nginx
+:app: nginx
+:group: demo
+endif::[]
+
+[id="osdk-run-proxy_{context}"]
+
+= Enabling proxy support
+
+To support proxied clusters, your Operator must inspect the environment for the following standard proxy variables and pass the values to Operands:
+
+- `HTTPS_PROXY`
+- `HTTPS_PROXY`
+- `NO_PROXY`
+
+.Procedure
+ifdef::golang[]
+. Add the `proxy.ReadProxyVarsFromEnv` helper function to the reconcile loop in the `controllers/memcached_controller.go` file and append the results to the Operand environments:
++
+[source,golang,subs="attributes+"]
+----
+...
+for i, container := range dep.Spec.Template.Spec.Containers {
+		dep.Spec.Template.Spec.Containers[i].Env = append(container.Env, proxy.ReadProxyVarsFromEnv()...)
+}
+...
+----
+
+endif::[]
+
+ifdef::ansible[]
+. Add the environment to the deployment by updating the `roles/memcached/tasks/main.yml` file with the following:
++
+[source,yaml,subs="attributes+"]
+----
+...
+env:
+   - name: HTTP_PROXY
+     value: '{{ lookup("env", "HTTP_PROXY") | default("", True) }}'
+   - name: http_proxy
+     value: '{{ lookup("env", "HTTP_PROXY") | default("", True) }}'
+...
+----
+
+endif::[]
+
+ifdef::helm[]
+. To include the overrides based on an environment variable, modify the `watches.yaml` file to include the 'overrideValues' field:
++
+[source,yaml,subs="attributes+"]
+----
+- group: demo.example.com
+  version: v1alpha1
+  kind: Nginx
+  chart: helm-charts/nginx
+  overrideValues:
+    proxy.http: $HTTP_PROXY
+#+kubebuilder:scaffold:watch
```

Comment:
This doesn't look like YAML, is it purposeful or should it be removed?

Reply:
This is a tag created by Kubebuilder scaffolding that indicates where future watches can be added. This is not strictly necessary for these docs, but as-is this shows the full watches.yaml rather than an abbreviation. IMO we are ok to include or not, either way.

Reply:
Okay that's fine then, thanks for confirming!